### ISSUE TITLE
[Feat] Provide scipy.ndimage.measurements and .morphology functions

### ISSUE THREAD

**Issue by elistevens** [kornia/kornia#428]
It would be nice if there were more performant versions of the functions in https://docs.scipy.org/doc/scipy-0.16.0/reference/ndimage.html#module-scipy.ndimage.measurements

For my use case, backpropagation isn't needed.

I'm mostly interested in `label`, `center_of_mass`, and the binary erosion, dilation, hole filling, etc.

Here's what I have been using (note that these are not exact reimplementations; they just work for my use case):

```python
class MorphologySubstitute(nn.Module):
    def __init__(self):
        super().__init__()

        self.conv_list = nn.ModuleList([
            self._make_circle_conv(radius) for radius in range(1, 8)
        ])

    def _make_circle_conv(self, radius):
        diameter = 1 + radius * 2

        a = torch.linspace(-1, 1, steps=diameter)**2
        b = (a[None] + a[:, None])**0.5

        circle_weights = (b <= 1.0).to(torch.float32)

        conv = nn.Conv2d(1, 1, kernel_size=diameter, padding=radius, bias=False)
        conv.weight.data.fill_(1)
        conv.weight.data *= circle_weights / circle_weights.sum()

        return conv


    def erode(self, input_mask, radius, threshold=1):
        conv = self.conv_list[radius - 1]
        input_float = input_mask.to(torch.float32)
        result = conv(input_float)

        return result >= threshold

    def dilate(self, input_mask, radius, threshold=0):
        conv = self.conv_list[radius - 1]
        input_float = input_mask.to(torch.float32)
        result = conv(input_float)

        return result > threshold

    # Note that this is NOT fill_holes, since it doesn't handle complex boundaries well
    def fill_cavity(self, input_mask):
        cumsum = input_mask.cumsum(-1)
        filled_mask = (cumsum > 0)
        filled_mask &= (cumsum < cumsum[..., -1:])
        cumsum = input_mask.cumsum(-2)
        filled_mask &= (cumsum > 0)
        filled_mask &= (cumsum < cumsum[..., -1:, :])

        return filled_mask
```

**Comment by edgarriba** [comment]
@elistevens thanks for this small contribution :) 

we definitely have to work on the morphology module for the next release.
There is an open discuss about it. I think we need to find an agreement in terms of API design for soft vs non-soft implementation.

Check this: https://github.com/kornia/kornia/pull/292  and https://github.com/kornia/kornia/issues/293

**Comment by edgarriba** [comment]
@elistevens morphology is in work in progress. What about measurements ? Could you open an ticket reporting what do you think it makes sense to have ?

**Comment by elistevens** [comment]
I'm not sure what I'd put in the ticket that's not already in this one. Can't you reopen this ticket? I don't have working implementations of label or center of mass, so I don't have much to contribute there.

**Comment by edgarriba** [comment]
@elistevens we included a morphology module. Please, check it out :)
https://kornia.readthedocs.io/en/latest/morphology.html

**Comment by elistevens** [comment]
Thanks!  Do you think that label and center of mass make sense?

**Comment by edgarriba** [comment]
from what I read, `label` would be the classical connected components algorithm, right ? can this be differentiable ?

As for center of mass, I believe that we already have some sort of approximation that could be used as a base:
https://kornia.readthedocs.io/en/latest/geometry.subpix.html#kornia.geometry.subpix.spatial_expectation2d
or
https://kornia.readthedocs.io/en/latest/geometry.subpix.html#kornia.geometry.subpix.conv_soft_argmax2d
